cmake/macros/macro_deal_ii_add_test.cmake

```
@@ -365,9 +369,17 @@ FUNCTION(DEAL_II_ADD_TEST _category _test_name _comparison_file)
           ${TEST_LIBRARIES} ${TEST_LIBRARIES_${_build}}
           )
 
+        SET(_performance_test_macro_definition "")
+        IF(ENABLE_PERFORMANCE_TESTS)
+          SET(_performance_test_macro_definition "ENABLE_PERFORMANCE_TESTS")
+        ENDIF()
+
         SET_PROPERTY(TARGET ${_target} APPEND PROPERTY
-          COMPILE_DEFINITIONS SOURCE_DIR="${CMAKE_CURRENT_SOURCE_DIR}"
+          COMPILE_DEFINITIONS
+            SOURCE_DIR="${CMAKE_CURRENT_SOURCE_DIR}"
+            ${_performance_test_macro_definition}
```

and here maybe pass
```suggestion
            ENABLE_PERFORMANCE_TESTS=${_performance_test_macro_definition}
```
to pass the semantics of the definition down.

---

But then I have the issue that I have to make sense of the value of `ENABLE_PERFORMANCE_TESTS` in C++ code where `ENABLE_PERFORMANCE_TESTS` can be anything (quoting from the CMake documentation): "1, ON, YES, TRUE, Y, or a non-zero number (including floating point numbers)" and all mixed cases of that.

I know that the way how this macro is expanded is a bit idosyncratic (mainly to avoid another 5 lines all all-caps CMake code). I am happy to change this behavior to a separate call to set target properties if this makes it more readable.

---

Yes, I get it now based on your explanation above. Thanks.